src/client/ua_client_highlevel_subscriptions.c

```diff
@@ -49,14 +49,20 @@ UA_Client_Subscriptions_new(UA_Client *client, UA_SubscriptionSettings settings,
     return UA_STATUSCODE_GOOD;
 }
 
-/* remove the subscription remotely */
-UA_StatusCode
-UA_Client_Subscriptions_remove(UA_Client *client, UA_UInt32 subscriptionId) {
-    UA_Client_Subscription *sub;
+static UA_Client_Subscription *findSubscription(UA_Client *client, UA_UInt32 subscriptionId)
```

Comment:
Nice one! One small change request:
Can you use const wherever possible, i.e., change this (if the compiler agrees) to 
`static UA_Client_Subscription *findSubscription(const UA_Client *client, const UA_UInt32 subscriptionId)`

Thanks!